### PR TITLE
Merge 1.5 into master

### PR DESCRIPTION
**Pull request by jjanvier**

### CODE REVIEW

**Comment by ClementGautier** [comment]
GTM

**Comment by solivier** [comment]
gtm

**Comment by nidup** [comment]
GTM

File: reference/performances_guide/rules_execution_memory_usage.rst [hunk 1]
```diff
@@ -71,3 +71,14 @@ So remember to fine tune the page size parameters if you encounter this kind of
...
+Memory leak fix in Rules Engine (ORM)
+-------------------------------------
+
+We discovered a memory leak occurring during the execution of rules with Doctrine ORM. It appears on ``commit()`` after products have been loaded and possibly modified.
+
+Our product entities tracking policy is set to "DEFFERED_EXPLICIT". It means that we are responsible for noticing Doctrine when a change has been made to an entity.
+When we notice Doctrine that an entity changed even when it has not, the library keeps a reference to this object internally.
+
+Unfortunately, when we try to ``flush()`` the entity manager, those objects are not detached. As we iterate and load more and more products, the number of references keeps growing until all memory is used.
+We decided to manually clean these unused references in ``Akeneo\Bundle\StorageUtilsBundle\Doctrine\Common\Detacher\ObjectDetacher`` to avoid this memory leak.
```
Comment (phaseinducer) [inline reference/performances_guide/rules_execution_memory_usage.rst]: We need to remove this part (and the fix) if the memory leak has been fixed after upgrading to Doctrine ODM 3.0.1.

**Pull request merged**